### ISSUE TITLE
from_remote_file feature

### ISSUE THREAD

**Issue by hilmiller** [f5devcentral/f5-puppet#48]
The types `f5_sslkey` and `f5_sslcertificate` have parameter `from_local_file`, but does this module provide a way to `PUT` files to the F5 from Puppet over iControl REST API?

**Comment by scshitole** [comment]
@hilmiller you mean from Puppet master to F5 bigip?

**Comment by hilmiller** [comment]
@scshitole Yeah, puppet master has the cert and the puppet device run puts the cert on the f5. I don't think that's possible currently. Perhaps a better solution would be to create a "f5_put_file" type that could handle uploading arbitrary F5 configuration objects. This way, more complex certificate chaining configurations could be implemented that simply expect local certificate files. The user would need to specify certificate related resources depend on f5_put_file resources running first.

**Comment by scshitole** [comment]
@hilmiller you are saying basically just the opposite of https://github.com/f5devcentral/f5-puppet/blob/master/README.md#f5_sslcertificate

**Comment by hilmiller** [comment]
Yeah, thanks!
